ee/kernel/include/timer_alarm.h

```diff
@@ -0,0 +1,61 @@
+/*
+# _____     ___ ____     ___ ____
+#  ____|   |    ____|   |        | |____|
+# |     ___|   |____ ___|    ____| |    \    PS2DEV Open Source Project.
+#-----------------------------------------------------------------------
+# Copyright 2022, ps2dev - http://www.ps2dev.org
+# Licenced under Academic Free License version 2.0
+# Review ps2sdk README & LICENSE files for further details.
+*/
+
+#ifndef __TIMER_ALARM_H__
+#define __TIMER_ALARM_H__
+
+#include <tamtypes.h>
+
+struct timer_alarm_t;
+typedef void (*timer_alarm_callback_t)(struct timer_alarm_t *alarm, void *arg);
+
+// Defines an alarm object
+// Do not manipulate fields directly, but through functions.
+
+struct timer_alarm_t {
+    u64 scheduled_time;                  // Clock tick when the alarm should be triggered
+    u64 timer_cycles;                    // Alarm period in clock ticks
+    struct timer_alarm_t *prev, *next;   // Previous and next alarm in the sorted event list
+    timer_alarm_callback_t callback;     // User callback to call during the IRQ
+    void *usr_arg;
+};
+
+// Init/Deinit functions
+void InitTimerAlarm();
+void DeinitTimerAlarm();
+
+// Allocate a timer-based alarm
+void InitializeTimerAlarm(struct timer_alarm_t *alarm);
+// Sets up an alarm (with a user callback function and argument) that triggers some cycles in the future.
+void SetTimerAlarm(struct timer_alarm_t *alarm, u64 clock_cycles, timer_alarm_callback_t user_callback, void *user_arg);
+// Starts/Enables the alarm.
+void iStartTimerAlarm(struct timer_alarm_t *alarm);
+void StartTimerAlarm(struct timer_alarm_t *alarm);
+// Stops/Disabled the alarm.
+void iStopTimerAlarm(struct timer_alarm_t *alarm);
+void StopTimerAlarm(struct timer_alarm_t *alarm);
+// Sleeping function
+void ThreadWaitClock(u64 clock_cycles);
+
+// Conversion functions, from time to bus cycles
+static inline u64 USecToCycles(u64 usec) {
+    return (147456 * usec) >> 10;  // approximate
+}
+
+static inline u64 MSecToCycles(u64 msec) {
+    return 147456 * msec;
```

Comment:
there are constans already defined for these values `kBUSCLK` and family in `ee/kernel/include/timer.h`

Reply:
Indeed!